modules/iap/iap.tf

```diff
@@ -21,6 +21,23 @@ data "google_project" "project" {
   project_id = var.project_id
 }
 
+## Check if domain and members_allowlist is provided
+data "null_data_source" "domain_validation" {
+  inputs = {}
+
+  lifecycle {
+    precondition {
+      condition     = var.domain != ""
+      error_message = "IAP configured requires domain name, Please provide a valid domain name for ${var.app_name} application."
+    }
+
+    precondition {
+      condition     = length(var.members_allowlist) != 0
+      error_message = "IAP configured requires valid members_allowlist, Please provide a valid input for ${var.app_name} application."
```

Comment:
"IAP configuration requires allowlisting users. Please provide a valid allowlist for ${var.app_name} application."